elyra/pipeline/component.py

```diff
@@ -223,133 +232,179 @@ def _log_warning(msg: str, logger: Optional[Logger] = None):
             print(f"WARNING: {msg}")
 
 
-class ComponentCategory(object):
+class ComponentReader(LoggingConfigurable):
     """
-    Represents a category assigned to a component
+    Abstract class to model component_entry readers that can read components from different locations
     """
+    location_type: str = None
 
-    def __init__(self, id: str, label: Optional[str],
-                 image_location: Optional[str],
-                 description: Optional[str]):
-        """
-        :param id: A unique identifier for the category
-        :param label: A ui-friendly label for the category
-        :param image_location: TODO Add info here once image serving details are decided
-        :param description: A description for the category
-        """
-
-        if not id:
-            raise ValueError("Invalid component category: Missing field 'id'.")
-
-        self._id = id
-        self._label = label
-        self._image_location = image_location
-        self._description = description
-
+    '''
     @property
-    def id(self) -> str:
-        return self._id
+    def location_type(self) -> str:
+        return self.location_type
+    '''
 
     @property
-    def label(self) -> str:
-        if not self._label:
-            return self._id
-        return self._label
+    def resource_type(self):
+        return self.location_type
 
-    @property
-    def image_location(self) -> Optional[str]:
-        return self._image_location
+    @abstractmethod
+    def read_component_definition(self, location: str) -> Optional[str]:
+        """
+        Read an absolute location to get the contents of a component
+        specification file
+        """
+        raise NotImplementedError()
 
-    @property
-    def description(self) -> Optional[str]:
-        return self._description
+    @abstractmethod
+    def get_list_of_paths(self, paths: List[str], parser_file_types: List[str]) -> List[str]:
```

Comment:
Do `parser_file_types` ever come into play?  ~I didn't catch their use in the existing implementations.~

`parser_file_types` appear to only apply to directory readers (and perhaps GitHub once implemented).  Should these values be fed into the appropriate constructors when the reader instance is instantiated?  Are these values static between a given component parser instance and its reader?  I think it would be good to remove these from the abstract method definition unless they apply to all instances.

Also, since "path" is very filesystem-ish, I wonder if a name like `get_absolute_locations` might be more generic since it could also apply to URLs?  It would take a list of "potentially relative" locations and return a list of absolute locations.

The directory-based readers would then apply the file-types filters that were provided at their construction.